test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -8400,6 +8402,31 @@ def cleanup_jobs_and_reservations(self, extend='forcedeletehist'):
         self.cleanup_reservations()
         return rv
 
+    def cleanup_jobs_for_perftests(self, runas=None):
+        """
+        Helper function to cleanup jobs in performance tests that deal with
+        large number of jobs. Job processes are killed manually to speed up
+        cleanup in performance tests
+        """
+        self.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+        job_ids = self.select()
+
+        for jid in job_ids:
+            jobs = self.status(JOB, id=jid)
+            for job in jobs:
+                if 'session_id' in job:
+                    qstat = self.status(JOB, [ATTR_session], jid)
```

Comment:
since we are checking the status of each job I am not very sure if we are seeing any performance improvements due to this change. I get that it will not timeout and mom will be more stable.
Can you run some tests to show the performance numbers? for example, total 30k jobs with all 30k jobs in Q state, 30k jobs with 10k jobs in R state and 20k in Q state, etc ...

Reply:
Maybe it would be faster to do one status() and get all the jobs in one go.  This means when we're deleting 20k we only make one expensive call to status() instead of 20k less expensive calls to status().

Reply:
I have changed code to get status in one go.

Reply:
Some numbers from the tests that I ran:

  | With changes | Existing
-- | -- | --
10K running jobs | 37.8 minutes | qdel operation hanged.noresponse for close to 1 hr
10K Queued jobs | 26.16 minutes | 27.16 minutes
10K jobs (5k Running and  5k Queued) | 36.8 minutes | qdel operation hanged.noresponse for close to 1 hr
5000R | 25 mins | qdel operation hanged.noresponse for close to 1 hr
5000Q | 14.8 mins | 14.6
1000R | 2.96 | 10.26
500R 500Q | 1.2 | 4.56

Reply:
Thanks @latha-subramanian for posting these numbers. I believe the solution is more stable than faster for large jobs, could be because underlying PBS qdel is slow. 
Could you post these numbers under testing section as well for the future record.